src/circuit/gadgets/constraints.rs

```diff
@@ -804,13 +803,26 @@ pub(crate) fn allocate_is_negative<F: LurkField, CS: ConstraintSystem<F>>(
 mod tests {
     use super::*;
 
-    use bellperson::util_cs::test_cs::TestConstraintSystem;
+    use bellperson::{util_cs::test_cs::TestConstraintSystem, Index, Variable};
     use blstrs::Scalar as Fr;
+    use ff::Field;
     use proptest::prelude::*;
     use std::ops::{AddAssign, SubAssign};
 
     use crate::field::FWrap;
 
+    // Strategy for creating a random linear combination of 5 vars
+    fn random_lc_strategy() -> impl Strategy<Value = LinearCombination<Fr>> {
+        prop::collection::vec(any::<FWrap<Fr>>(), 5).prop_map(|coeffs| {
+            let mut lc = LinearCombination::zero();
+            for coeff in coeffs {
+                let var = Variable::new_unchecked(Index::Input(rand::random()));
```

Comment:
I think just allocating variables sequentially would be less confusing than creating them with random indexes in the `usize` range.